In [2]:
import pandas as pd

# class 1 

In [4]:
df = pd.read_excel("50_startups_messy.xlsx")
print(df.shape)

(50, 5)


In [5]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 50 entries, 0 to 49
Data columns (total 5 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   R&D Spend        50 non-null     float64
 1   Administration   50 non-null     float64
 2   Marketing Spend  49 non-null     object 
 3   State            50 non-null     str    
 4   Profit           50 non-null     float64
dtypes: float64(3), object(1), str(1)
memory usage: 2.1+ KB


2.R&D Spend = continous
  Administration = continous
  Marketing spend = nominal
  State = nominal
  Profit = continous
3.Marketing spend has one missing value (49 non-null) , I would handle this by filling the missing value with the average (mean) of the rest of the Marketing Spend column so we don't lose the other data in that row.

In [7]:
series_example =  df["State"]
df[["Administration"]]
# 1. Print the type of your Series
print(type(series_example))

# 2. Print the shape of your Series
print(series_example.shape)

# 3. Print the type of your DataFrame
print(type(df[["Administration"]]))

# 4. Print the shape of your DataFrame
print(df[["Administration"]].shape)

<class 'pandas.Series'>
(50,)
<class 'pandas.DataFrame'>
(50, 1)


In [8]:
df.describe()

,R&D Spend,Administration,Profit
count,50.000000,50.000000,50.000000
mean,73721.615600,121344.639600,112012.639200
std,45902.256482,28017.802755,40306.180338
min,0.000000,51283.140000,14681.400000
25%,39936.370000,103730.875000,90138.902500
50%,73051.080000,122699.795000,107978.190000
75%,101602.800000,144842.180000,139765.977500
max,165349.200000,182645.560000,192261.830000


MEAN VALUES : 
R&D Spend = 73721.615600
Administration = 121344.639600
Profit = 112012.639200
Marketing spend was not recorded as numerical values , it's data type was object

In [10]:
df.to_json('startups_records.json', orient='records')

6.When exported to Json all values formerly as rows appeared as a single list on one column
7.The university could take record of a student's Height, Address and Age
Height; measurement of physical dimension
Address : Measurement of geographical location
Age : chronological age of the student 

# Class 2

In [11]:
age = [19,22,21,35,28,24,31,19,27,40]
print(len(age))

10


In [12]:
total_age = sum(age)
print (total_age)

266


In [13]:
mean_age = total_age / len(age)
print (mean_age)

26.6


In [14]:
min_age = min(age)
print (min_age)
max_age = max(age)
print (max_age)

19
40


In [19]:
import statistics
median_age = statistics.median(age)
print (median_age)

25.5


Yes the mean 26.6 is different from the median 25.5 ,when the mean and the median disagree, it tells you that the data distribution is skewed rather than perfectly symmetrical


# class 3

1.  what is Data science : Data science is a multidisciplinary field combining statistics and mathematics, programming, and domain knowledge to clean, process, analyze, and interpret large amounts of data
2. Name the seven stages of the data science lifecycle :
. Problem definition
. Data collection
. Data Preparation
. Exploration 
. Analysis & Modelling
. Evaluating & Monitoring
. Deployment
3. Which stage usually consumes time and why : Stages 2, 3, and 4 (Data collection, data preparation, and exploration) routinely consume the most time because real-world data is unstructured and requires extensive cleaning and formatting before it can actually be analyzed
4. Name a data science application sector you care about : Under RETAIL AND E COMMERCE
 Recommendation engines (such as "customers also bought" suggestions) that analyze user clicks, purchases, and browsing history to personalize the shopping experience.
